.github/workflows/pinot_tests.yml

```diff
@@ -123,6 +123,11 @@ jobs:
             --add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED
             --add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED
             --add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED
+            --add-opens=java.base/java.nio=ALL-UNNAMED
+            --add-opens=java.base/sun.nio.ch=ALL-UNNAMED
+            --add-opens=java.base/java.lang=ALL-UNNAMED
+            --add-opens=java.base/java.util=ALL-UNNAMED
+            --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
```

Comment:
@ankitsultana I was able to fix the unit test failures on JDK21 by adding these exceptions.

Sorry for the commit spam, for some reason Github CI behavior is different than my local setup so it took a while to get to the correct config.